### PR TITLE
Fix chain name outside viewport

### LINKED ISSUE #714: Copy button moved out of the viewport, if the name is to long
Copied

### PR DESCRIPTION
**Pull request by imollov**
fixes #714
depends on #776

recording

### CODE REVIEW

File: src/styles/global.scss
Comment (mradkov) [inline src/styles/global.scss:16]: Isn't this conflicting with #776 ?

**Review by davidyuk — APPROVED** [review]
The code changes look good, but I'm not sure that the way long names showed is user-friendly. I would consider
- to hide the middle of the name, like `tttttttt...ttt.chain`
- or to implement an analog of marquee tag 😄 (maybe on hover)

cc: @onvisions

File: src/styles/global.scss
Comment (davidyuk) [reply src/styles/global.scss:16]: this is the same change, git should handle this well

Comment (imollov) [reply src/styles/global.scss:16]: I needed to do the same change as part of the fix. My intention is to rebase and remove the duplicated commit once #776 is merged. 

I've also added it as depending to that PR in the description. Please let me know if there's a more appropriate way to handle this case.

**Comment by onvisions** [comment]
@davidyuk 
I discussed it already with @mradkov. What @imollov did was synced with me. It's a temporary design fix. I think it's the better option at the moment until we have special design for displaying very long names. I'll do this as part of UX/UI wallet rebranding and improvement I'm currently working on.

The problem with your suggestion above would be that:
1. on mobile we don't have hover interactions. This means that the user will have to click to show the full name "tag" but the .chain name element is **supposed to be link to the user account** (opened in blockchain explorer).
2. still there might be not enough space to display really long names within a tag or something...There is no max limit of names characters so virtually any length is possible. **Something to discuss about this** is whether we really need limitless .chain names as this is probably not very useful to anybody.

**Review by onvisions — COMMENTED** [review]
@davidyuk
I discussed it already with @mradkov. What @imollov did was synced with me. It's a temporary design fix. I think it's the better option at the moment until we have special design for displaying very long names. I'll do this as part of UX/UI wallet rebranding and improvement I'm currently working on.

The problem with your suggestion above would be that:

   

1.  on mobile we don't have hover interactions. This means that the user will have to click to show the full name "tag" but the .chain name element is supposed to be link to the user account (opened in blockchain explorer).
2.   still there might be not enough space to display really long names within a tag or something...There is no max limit of names characters so virtually any length is possible. Something to discuss about this is whether we really need limitless .chain names as this is probably not very useful to anybody.

**Comment by davidyuk** [comment]
> There is no max limit of names characters so virtually any length is possible.

Actually, AENS documentation says:

> Names SHOULD be at most 253 characters long, in order to enable interoperability with DNS.

https://aeternity.com/documentation-hub/protocol/AENS/

**Comment by mradkov** [comment]
> > There is no max limit of names characters so virtually any length is possible.
> 
> Actually, AENS documentation says:
> 
> > Names SHOULD be at most 253 characters long, in order to enable interoperability with DNS.
> 
> https://aeternity.com/documentation-hub/protocol/AENS/

@imollov @onvisions Lets come up with a nice UX solution taking into account the limit of 253 chars. I doubt people would be having such long names anyways but still.

I'm will merge this PR now, and once we have a better UX suggestion we can refactor it.

**Pull request merged**
